test/integration/targets/nxos_telemetry/tests/common/merged.yaml

```diff
@@ -0,0 +1,122 @@
+          - { id: 55, data_source: DME, path: {name: sys/ospf, depth: 0, query_condition: foo, filter_condition: 'or(eq(ethpmPhysIf.operSt,"down"),eq(ethpmPhysIf.operSt,"up"))'}}
+        subscriptions:
+          - { id: 44, destination_group: 10, sensor_group: {id: 8, sample_interval: 2000}}
+          - { id: 44, destination_group: 2, sensor_group: {id: 2, sample_interval: 2000}}
+          - { id: 55, destination_group: 10, sensor_group: {id: 55, sample_interval: 2000}}
+      provider: "{{ connection }}"
+    register: result
+
+  - assert:
+      that:
+        - "(ansible_facts.network_resources.telemetry|dict2items)|symmetric_difference(result.before|dict2items)|length == 0"
+
+  - name: Gather Telemetry Facts After Changes
+    nxos_facts: *facts
+
+  - assert:
+      that:
+        - "(ansible_facts.network_resources.telemetry|dict2items)|symmetric_difference(result.after|dict2items)|length == 0"
+
+  - name: Telemetry subscription - merged - idempotence
+    nxos_telemetry: *merged
+    register: result
+
+  - assert:
+      that:
+        - "result.changed == false"
+
+  - name: Telemetry subscription - change values
+    nxos_telemetry: &merged_change
+      state: 'merged'
+      config:
+        certificate:
+          key: /bootflash/local_server.key
+          hostname: localhost
+        compression: gzip
+        source_interface: Loopback77
+        vrf: management
+        destination_groups:
+          - id: 2
+            destination:
+              ip: 192.168.0.1
+              port: 50001
+              protocol: grpc
+              encoding: gpb
+          - { id: 2, destination: {ip: 192.168.0.2, port: 60001, protocol: gRPC, encoding: GPB}}
+          - { id: 10, destination: {ip: 192.168.0.1, port: 50001, protocol: Grpc, encoding: gPB}}
+          - { id: 10, destination: {ip: 192.168.0.2, port: 60001, protocol: gRPC, encoding: gpb}}
+        sensor_groups:
+          - { id: 8, data_source: NX-API, path: {name: sys/bgp, depth: 0, query_condition: foo, filter_condition: foo}}
+          - { id: 2, data_source: NX-API, path: {name: sys/bgp/inst, depth: unbounded, query_condition: foo, filter_condition: foo}}
+          - { id: 55, data_source: DME, path: {name: 'sys/bgp/inst/dom-default/peer-[10.10.10.11]/ent-[10.10.10.11]', depth: 0, query_condition: foo, filter_condition: foo}}
+          - { id: 55, data_source: DME, path: {name: sys/ospf, depth: 0, query_condition: foo, filter_condition: 'or(eq(ethpmPhysIf.operSt,"down"),eq(ethpmPhysIf.operSt,"up"))'}}
+        subscriptions:
+          - { id: 44, destination_group: 10, sensor_group: {id: 8, sample_interval: 1000}}
+          - { id: 44, destination_group: 2, sensor_group: {id: 2, sample_interval: 2000}}
+          - { id: 55, destination_group: 10, sensor_group: {id: 55, sample_interval: 2000}}
+      provider: "{{ connection }}"
+    register: result
+
+  - assert:
+      that:
+        - "result.changed == true"
```

Comment:
I've noticed that a lot of our tests are like this - they check for `changed == true` but with a large test like this it will pass even if only one property has changed. Should this test also assert for individual property changes?

Reply:
Good suggestion.  I added the following check to make sure the right commands are sent and only those commands.

```yaml
    # The step above should result in only the following changes:
    # "commands": [
    #     "telemetry",
    #     "certificate /bootflash/local_server.key localhost",
    #     "subscription 44",
    #     "snsr-grp 8 sample-interval 1000"
    # ],

  - set_fact:
      test_list:
        - "telemetry"
        - "certificate /bootflash/local_server.key localhost"
        - "subscription 44"
        - "snsr-grp 8 sample-interval 1000"

  - assert:
      that:
        - "result.changed == true"
        - "test_list|symmetric_difference(result.commands)|length == 0"
```